File: app/src/main/java/me/jromero/connect2sql/adapter/TableListAdapter.java
package me.jromero.connect2sql.adapter;

import java.util.List;

import com.gitlab.connect2sql.R;
import me.jromero.connect2sql.sql.Table;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

public class TableListAdapter extends BaseAdapter {
    private final List<Table> tableList;

    private final LayoutInflater mInflater;

    public TableListAdapter(Context context, List<Table> tableList) {
        this.tableList = tableList;
        notifyDataSetChanged();
        mInflater = LayoutInflater.from(context);
    }

    public void clear() {
        tableList.clear();
        notifyDataSetChanged();
    }

    public void add(Table table) {
        tableList.add(table);
        notifyDataSetChanged();
    }

    public List<Table> getTables() {
        return tableList;
    }

    @Override
    public int getCount() {
        return tableList.size();
    }

    @Override
    public Object getItem(int position) {
        return tableList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = mInflater.inflate(R.layout.item_table, null);
            holder = new ViewHolder();
            holder.image = (ImageView) convertView.findViewById(R.id.imageView1);
            holder.txtName = (TextView) convertView.findViewById(R.id.text1);
            holder.txtType = (TextView) convertView.findViewById(R.id.text2);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        Table table = tableList.get(position);
        String name = table.getName();
        String typeString = "";
        if (table.getType() == Table.TYPE_VIEW) {
            typeString = "View";
            holder.image.setImageResource(R.drawable.icon_view);
        } else {
            typeString = "Table";
            holder.image.setImageResource(R.drawable.icon_table);
        }

        holder.txtName.setText(name);
        holder.txtType.setText("(" + typeString + ")");
        return convertView;
    }

    private static class ViewHolder {
        ImageView image;
        TextView txtName;
        TextView txtType;
    }

}

File: app/src/main/java/app/devlife/connect2sql/adapter/TableListAdapter.kt
package app.devlife.connect2sql.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import app.devlife.connect2sql.sql.Table
import com.gitlab.connect2sql.R

class TableListAdapter(private val context: Context, private val tableList: MutableList<Table>) : BaseAdapter() {

    private val mInflater: LayoutInflater

    val tables: List<Table>
        get() = tableList

    init {
        notifyDataSetChanged()
        mInflater = LayoutInflater.from(context)
    }

    fun clear() {
        tableList.clear()
        notifyDataSetChanged()
    }

    fun add(table: Table) {
        tableList.add(table)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return tableList.size
    }

    override fun getItem(position: Int): Any {
        return tableList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = convertView ?: mInflater.inflate(R.layout.item_table, parent).apply {
            this.tag = ViewHolder(
                this.findViewById(R.id.imageView1),
                this.findViewById(R.id.text1),
                this.findViewById(R.id.text2)
            )
        }

        val holder: ViewHolder = view.tag as ViewHolder

        with(tableList[position]) {
            holder.txtName.text = this.name
            when (this.type) {
                Table.TYPE_TABLE -> {
                    holder.image.setImageResource(R.drawable.icon_table)
                    holder.txtType.text = context.getString(R.string.table_list_type_table)
                }
                Table.TYPE_VIEW -> {
                    holder.image.setImageResource(R.drawable.icon_view)
                    holder.txtType.text = context.getString(R.string.table_list_type_view)
                }
            }
        }

        return view
    }

    private data class ViewHolder(
        val image: ImageView,
        val txtName: TextView,
        val txtType: TextView
    )
}
